src/DonorPerfect.php

```diff
@@ -121,10 +121,12 @@ protected function callInternal(array $params)
 
         // Validate the API call before making it
         $relativeUrl .= http_build_query($args, null, '&', PHP_QUERY_RFC3986);
+        // encode any + signs
+        $relativeUrl = str_replace('+', '%2B', $relativeUrl);
```

Comment:
Could you provide the exact input / call that you were making that was causing this issue with the plus sign not being properly encoded?

Reply:
Hmm, I'll need you to test this some more, I ran a simplified version of this to test and it seems to be encoding the `+` correctly, see below:

```php
/**
 - dp_savedonor
    - prepareParams
      - prepareString
    - call
 */

function call(string $action, array $parameters)
{
    $paramString = '';
    $i = 0;
    foreach ($parameters as $param => $value) {
        $original = $value;
        $value = trim($value);

        if (
            !is_object($original)
            && is_numeric($value)
            && strpos($value, 'e') === false
            && strpos($value, '+') === false
        ) {
            $value = $value;
        } elseif (is_bool($value)) {
            $value = $value ? '1' : '0';
        } elseif (empty($value)) {
            $value = 'NULL';
        } elseif (is_array($value)) {
            $value = "N'" . implode($value, '|') . "'";
        } else {
            // Ensure quotes are doubled for escaping purposes
            // @see https://api.warrenbti.com/2020/08/03/apostrophes-in-peoples-names/
            $value = str_replace(["'", '"', '%'], ["''", '', '%25'], $value);

            // Wrap the value in quotes
            $value = "'$value'";
        }

        $paramString .= '@'.$param.'='.$value;
        ++$i;
        if ($i !== count($parameters)) {
            $paramString .= ',';
        }
    }

    $params = [
        'action' => $action,
        'params' => $paramString,
    ];

    return $params;
}


$params = [
    'mobile_phone' => new class('+19028307662') {
            protected string $value;
            public function __construct(string $value)
            {
                $this->value = $value;
            }
            public function __toString()
            {
                return $this->value;
            }
        },
];

$prepared = call('dp_savedonor', $params);

$result = http_build_query($prepared, null, '&', PHP_QUERY_RFC3986);
```

Which generates `action=dp_savedonor&params=%40mobile_phone%3D%27%2B19028307662%27`